src/app/features/rpc-request/rpc-request.layout.tsx

```diff
@@ -0,0 +1,86 @@
+import { Outlet } from 'react-router-dom';
+
+import { Box } from 'leather-styles/jsx';
+
+import type { Money } from '@leather.io/models';
+import { Approver } from '@leather.io/ui';
+import { baseCurrencyAmountInQuote, i18nFormatCurrency } from '@leather.io/utils';
+
+import { closeWindow } from '@shared/utils';
+
+import type { HasChildren } from '@app/common/has-children';
+import { ApproveTransactionError } from '@app/components/approve-transaction/approve-transaction-error';
+import { ApproveTransactionHeader } from '@app/components/approve-transaction/approve-transaction-header';
+import { ApproveTransactionActionsTitle } from '@app/components/approve-transaction/approve-transaction-title';
+import { ApproveTransactionWrapper } from '@app/components/approve-transaction/approve-transaction-wrapper';
+import { getApproveTransactionActions } from '@app/components/approve-transaction/get-approve-transaction-actions';
+import { BackgroundOverlay } from '@app/components/loading-overlay';
+import { useFeeEditorContext } from '@app/features/fee-editor/fee-editor.context';
+import { useRpcRequestContext } from '@app/features/rpc-request/rpc-request.context';
+import { useUiState } from '@app/store/ui/ui.slice';
+
+interface RpcRequestLayoutProps extends HasChildren {
+  title: string;
+  totalSpend: Money | null;
+  onApprove(): Promise<void>;
+}
+export function RpcRequestLayout({
```

Comment:
This name suggests it applies to all RPC requests, yet it contains info like `amount`, which is specific only to certain rpc requests. 

Either this should be more specific, `RpcTransactionRequestLayout`, or the more specific parts removed. Consider sign message for example, that fits in this naming, yet has no amount.